File: ros2_ws/src/my_python_pkg/my_python_pkg/smartphone.py
#!/usr/bin/env python3
from example_interfaces.msg import String

import rclpy
from rclpy.node import Node
 
 
class SmartphoneNode(Node):
    def __init__(self):
        super().__init__("smartphone")
        self.smartphone_name_ = "Smartphone"
        self.subscriber_ = self.create_subscription(String, "robot_news", self.robot_news_callback,  10) # We are creating a subscriber that listens to the "robot_news" topic. Whenever a message is published on this topic, the news_callback function will be called with the message as an argument.
        self.get_logger().warning("Smartphone is Ready to Use...")
    
    # The callback function that will be called whenever a message is received on the "robot_news" topic.
    def robot_news_callback(self, msg: String):
        self.get_logger().info(f"{self.smartphone_name_} received news: {msg.data} from Publisher ---msg.publisher_name---") # # msg.publisher_name is not a standard field in the String message type, so it will not work unless you have modified the message definition to include a publisher field. If you want to log the publisher's name, you would need to include that information in the message data or use a custom message type that includes a publisher field.
 
 
def main(args=None):
    rclpy.init(args=args)
    node = SmartphoneNode()
    rclpy.spin(node)
    rclpy.shutdown()
 
 
if __name__ == "__main__":
    main()

File: ros2_ws/src/my_cpp_pkg/src/smartphone.cpp
#include "rclcpp/rclcpp.hpp"
#include "example_interfaces/msg/string.hpp"


using namespace std::chrono_literals;
using namespace std::placeholders; 
// This is needed to use std::bind with member functions. It allows us to use _1, _2, etc. as placeholders for the arguments that will be passed to the callback function when it is called by the ROS 2 executor.

using String = example_interfaces::msg::String;
 
class SmartphoneNode : public rclcpp::Node  
{
public:
    SmartphoneNode() : Node("smartphone"), smartphone_name_("Smartphone")  {   
        this->subscriber_ = this->create_subscription<String>(
            "robot_news", 10, std::bind(&SmartphoneNode::robot_news_callback, this, _1)
        );
        RCLCPP_WARN(this->get_logger(), "Smartphone Node has started subscribing/listening to robot news [This time from the Hard C++ syntax :P]");
    }
 
private:
    std::string smartphone_name_;
    rclcpp::Subscription<String>::SharedPtr subscriber_;

    void robot_news_callback(const String::SharedPtr msg) {
        RCLCPP_ERROR(this->get_logger(), "%s received news: '%s'", smartphone_name_.c_str(), msg->data.c_str());
    }
    
};
 
int main(int argc, char **argv) {
    rclcpp::init(argc, argv);
    auto node = std::make_shared<SmartphoneNode>(); 
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}